bin/cucc.in

```diff
@@ -0,0 +1,287 @@
+                        help="Generate debug information for device code.")
+    # "The compiler has an option (-use_fast_math) that forces each
+    # function in Table 9 to compile to its intrinsic counterpart. "
+    # [CUDA PM v12.2 13.2].
+    #
+    # CUDA intrinsics have defined numerical accuracy. This means -ffast-math
+    # is potentially more relaxed than --use_fast_math so we may not use it.
+    parser.add_argument("-use_fast_math", "--use_fast_math",
+                        nargs=0, action=IgnoredOption)
+
+    return parser
+
+def get_hip_path() -> str:
+    """Get HIP path
+    """
+    return "@CUCC_HIP_PATH@"
+
+def get_hipcc() -> str:
+    """Get path to hipcc executable
+    """
+    return get_hip_path() + "/bin/hipcc"
+
+def get_cuda_include_dir() -> str:
+    """Return include directory for chipStar's CUDA headers.
+    """
+    return get_hip_path() + "/include/cuspv"
+
+def get_cuda_library_dir() -> str:
+    """Return directory for chipStar's CUDA libraries.
+    """
+    return get_hip_path() + "/lib"
+
+def determine_input_languages(arg_list : list[str], xmode=None) -> set[str]:
+    """Determine input language modes from the argument list
+
+    The 'xmode' specifies value of the last '-x' option if it was specified.
+    returns a set of languages (c, c++ and cu) involved.
+    """
+
+    # '-x' applies globally - unlike in other compilers usually.
+    if xmode is not None:
+        return xmode
+
+    # Scan over arguments and look for file extensions for determining
+    # input languages. This may give incorrect answer if unknown
+    # options with values are involved ('-opt-with-value silly.cu').
+    modes : set = set()
+    for arg in arg_list:
+        if len(modes) == 3: # All supported languages are present.
+            break
+
+        if arg.startswith("-"):
+            continue # An option, skip it.
+
+        _, ext = os.path.splitext(arg)
+
+        if not ext: # Perhaps, a value for an unknown option.
+            continue
+
+        if ext == ".c":
+            modes.add('c')
+        elif ext == ".cu":
+            modes.add('cuda')
+        elif ext in {'.cpp', '.cc'}:
+            modes.add('c++')
+
+    return modes
+
+def filter_args_for_hipcc(arg_list : list[str]) -> list[str]:
+    """Filter out arguments on the way to hipcc.
+    """
+    filtered = []
+    for arg in arg_list:
+        if arg == "-Xcompiler":
+            continue
+        if arg.startswith("-Xcompiler="):
+            arg = arg.replace("-Xcompiler=", "", 1)
+        filtered.append(arg)
+    return filtered
+
+def main():
+    """Driver implementation.
+    """
+    if os.environ.get('CHIP_CUCC_VERBOSE') == "1":
+        print(f"cucc args: {' '.join(map(quote, sys.argv))}",
+              file=sys.stderr)
+
+    known_args, other_args = prepare_argparser().parse_known_args()
+
+    if known_args.default_stream == "per-thread":
+        error_exit("per-thread stream function is not implemented.")
+
```

Comment:
Why not?

Reply:
It haven't been covered/tested/etc. thus can't assumed it's implemented.